### PR TITLE
Fix the issue that users cannot remove proxyUrl of notifier

### PR DESCRIPTION
**Pull request by GGGitBoy**
**Problem:**

After users set a proxy url for a notifier, they can not really remove it in alert manager config. 
Rancher will always set an empty proxyUrl in alert manger config for the notifier. So alert manager cannot send out alerts due to the empty proxy url setting.

**Solution:**

So we need to check if  it's empty.

**Related Issue:**

https://github.com/rancher/rancher/issues/26089

**Related PR:**

https://github.com/rancher/ui/pull/3854

### CODE REVIEW

File: pkg/controllers/user/alert/configsyncer/configsyncer.go [hunk 1]
```diff
@@ -556,7 +556,7 @@ func (d *ConfigSyncer) addRecipients(notifiers []*v3.Notifier, receiver *alertco
 					slack.Channel = r.Recipient
 				}
 
-				if notifier.Spec.SlackConfig.HTTPClientConfig != nil {
+				if notifier.Spec.SlackConfig.HTTPClientConfig != nil && notifier.Spec.SlackConfig.HTTPClientConfig.ProxyURL != "" {
```
Comment (gitlawr) [inline pkg/controllers/user/alert/configsyncer/configsyncer.go]: Given that this check happens many times I think we can have a utility function for it.
e.g.:
```
func IsHTTPClientConfigSet(cfg *v3.HTTPClientConfig) bool{}
```

**Comment by GGGitBoy** [comment]
Updated

File: pkg/notifiers/sender.go [hunk 2]
```diff
@@ -516,7 +516,7 @@ func NewClientFromConfig(cfg *v3.HTTPClientConfig, dialer dialer.Dialer) (*http.
 		Timeout: time.Second * 10,
 	}
 
-	if cfg != nil {
+	if cfg != nil && cfg.ProxyURL != "" {
```
Comment (gitlawr) [inline pkg/notifiers/sender.go]: The utility function can be used here as well. We can define it in `pkg/notifiers` and use it in `pkg/controllers`

File: pkg/notifiers/sender_test.go [hunk 3]
```diff
@@ -0,0 +1,29 @@
+package notifiers
+
+import (
+	"testing"
+
+	v3 "github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestIsHTTPClientConfigSet(t *testing.T) {
```
Comment (dramich) [inline pkg/notifiers/sender_test.go:10]: The tests should be setup with a list of cases to process, this helps make the tests easier to read and add additional tests later. An example can be seen here https://github.com/rancher/rancher/blob/master/pkg/rbac/common_test.go

**Comment by GGGitBoy** [comment]
Updated

**Comment by deniseschannon** [comment]
@GGGitBoy Can you make a backport PR into the release/v2.4 branch?

**Comment by GGGitBoy** [comment]
> @GGGitBoy Can you make a backport PR into the release/v2.4 branch?

release/v2.4 branch PR is here
https://github.com/rancher/rancher/pull/26393

**Pull request merged**